x-pack/plugins/cases/public/components/files/add_file.tsx

```diff
@@ -0,0 +1,151 @@
+import { FileUpload } from '@kbn/shared-ux-file-upload';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+
+import { constructFileKindIdByOwner } from '../../../common/files';
+import type { Owner } from '../../../common/constants/types';
+
+import { CommentType, ExternalReferenceStorageType } from '../../../common';
+import { FILE_ATTACHMENT_TYPE } from '../../../common/api';
+import { useCasesToast } from '../../common/use_cases_toast';
+import { useCreateAttachments } from '../../containers/use_create_attachments';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import * as i18n from './translations';
+import { useRefreshCaseViewPage } from '../case_view/use_on_refresh_case_view_page';
+
+interface AddFileProps {
+  caseId: string;
+}
+
+const AddFileComponent: React.FC<AddFileProps> = ({ caseId }) => {
+  const { owner, permissions } = useCasesContext();
+  const { client: filesClient } = useFilesContext();
+  const { showDangerToast, showErrorToast, showSuccessToast } = useCasesToast();
+  const { isLoading, createAttachments } = useCreateAttachments();
+  const refreshAttachmentsTable = useRefreshCaseViewPage();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = () => setIsModalVisible(true);
+
+  const onError = useCallback(
+    (error) => {
+      showErrorToast(error, {
+        title: i18n.FAILED_UPLOAD,
+      });
+    },
+    [showErrorToast]
+  );
+
+  const onUploadDone = useCallback(
+    async (chosenFiles: UploadedFile[]) => {
+      if (chosenFiles.length === 0) {
+        showDangerToast(i18n.FAILED_UPLOAD);
+        return;
+      }
+
+      const file = chosenFiles[0];
+
+      try {
+        await createAttachments({
+          caseId,
+          caseOwner: owner[0],
+          data: [
+            {
+              type: CommentType.externalReference,
+              externalReferenceId: file.id,
+              externalReferenceStorage: {
+                type: ExternalReferenceStorageType.savedObject,
+                soType: FILE_SO_TYPE,
+              },
+              externalReferenceAttachmentTypeId: FILE_ATTACHMENT_TYPE,
+              externalReferenceMetadata: {
+                files: [
+                  {
+                    name: file.fileJSON.name,
+                    extension: file.fileJSON.extension ?? '',
+                    mimeType: file.fileJSON.mimeType ?? '',
+                    created: file.fileJSON.created,
+                  },
+                ],
+              },
+            },
+          ],
+          updateCase: refreshAttachmentsTable,
+          throwOnError: true,
+        });
+
+        showSuccessToast(i18n.SUCCESSFUL_UPLOAD_FILE_NAME(file.fileJSON.name));
+      } catch (error) {
+        // error toast is handled inside  createAttachments
+
+        // we need to delete the file if attachment creation failed
+        await filesClient.delete({
```

Comment:
The problem here is that we want to delete **only the file** because this should run if the `createAttachment` call fails.

We only have `deleteFileAttachments` in the internal API right?

Reply:
> The problem here is that we want to delete only the file because this should run if the createAttachment call fails.

That's fine, we should still call our internal api.

> We only have deleteFileAttachments in the internal API right?
That's correct it's an internal api but should be accessible by the frontend.

Reply:
> Right, but doesn't deleteFileAttachment delete a file and an attachment? Here only the file will exist.

It should handle the scenario when only the file exists.
https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/server/client/attachments/bulk_delete.ts#L51-L56